### ISSUE TITLE
test/tcp sometimes stop on step 3: HUGE string

### ISSUE THREAD

**Issue by AndreasAakesson** [hioa-cs/IncludeOS#429]
Test *SOMETIMES* stops at `Step #3: HUGE string`.

VM (Virtualbox):
```
...
       [ TEST ] SMALL string (600)
                [x] conn.read() == small
                + Enabling IRQ 0, mask: 0xfbfa
[ IRQ manager ] Updated subscriptions: 0x401 irq: 0
       [ TEST ] BIG string (6000)
                [x] conn.read() == big
       [ TEST ] HUGE string (60000)
// stops here
```

Python:
```
$ python test.py 127.0.0.1 127.0.0.1
connecting to 127.0.0.1 port 8081
closing socket
connecting to 127.0.0.1 port 8082
closing socket
connecting to 127.0.0.1 port 8083
// stops here
```

If exiting the test, and start again, it works fine. 

Possible errors: 
* python script 
* TCP implementation 
* Buffer solution inside test service. `class Buffer`
* The event loop that fills/reads the buffer.
* My computer

Reproduce/Solution:
Start test and :pray: to RNG jesus

**Comment by AndreasAakesson** [comment]
This is probably due *congestion*. Currently improving TCP with **slow start** and **fast recovery**, hopefully the problem will be solved.

**Comment by AndreasAakesson** [comment]
This was due to checksum failure. Solved with 7eaeed6
